### ISSUE TITLE
Possible emoji ambiguity

### ISSUE THREAD

**Issue by gunnarhj** [mike-fabian/ibus-typing-booster#244]
I've just uploaded the latest version of `ibus-typing-booster` to Debian/Ubuntu. While the tests pass on both Debian's and Ubuntu's build servers, I noticed a failure with respect to `tests/test_emoji_candidates.py` when building locally not in a clean chroot, i.e. in an environment with a lot of installed packages which are irrelevant for the package build.

I attached [test-suite.log](https://github.com/mike-fabian/ibus-typing-booster/files/7158034/test-suite.log) with the complaints, and also [a patch](https://github.com/mike-fabian/ibus-typing-booster/files/7158043/skip-some-emoji-tests.patch.txt) which, if applied, works around the failures.

My thought is that this kind of confusion might happen when the program is used, and that's the reason why I submitted this issue for your consideration. I uninstalled packages with "emoji" in their names, including Noto Color Emoji, but that didn't help.

**Comment by mike-fabian** [comment]
The expected results of these tests depend very much on the CLDR annotations files used. As these change quite frequently, I often have to adapt the test results.

ibus-typing-booster includes these CLDR annotation files in its sources

https://github.com/mike-fabian/ibus-typing-booster/tree/main/data/annotations
https://github.com/mike-fabian/ibus-typing-booster/tree/main/data/annotationsDerived

I update these frequently from git master: https://github.com/unicode-org/cldr.git 

If these tests are run against other versions of these CLDR annotation files, it is likely that there will be failures.

On Fedora, there is a package `cldr-emoji-annotation-39-1.fc34.noarch` which contains the annotation files at `/usr/share/unicode/cldr/common`:

```
/usr/share/unicode/cldr/common/annotations
/usr/share/unicode/cldr/common/annotations/af.xml
…
```

and when building the ibus-typing-booster.rpm on Fedora, I don’t include the CLDR emoji annotations from the ibus-typing-booster sources in the ibus-typing-booster.rpm.

I.e. these are just empty directories in the ibus-typing-booster.rpm on Fedora:

```
$ rpm -ql ibus-typing-booster | grep annotations
/usr/share/ibus-typing-booster/data/annotations
/usr/share/ibus-typing-booster/data/annotationsDerived
mfabian@taka:~
$ 
```

This saves some space because ibus uses this CLDR emoji annotation data as well and uses thouse from the `cldr-emoji-annotation-39-1.fc34.noarch` package which therefore must be installed.

The annotations in this package are usually far older than those included in ibus-typing-booster, so using them saves some space but usually it is an older version. That doesn’t matter all that much, only the matches change slightly which might make test cases fail.

ibus-typing-booster looks in the following directories for CLDR emoji annotations files:

https://github.com/mike-fabian/ibus-typing-booster/blob/main/engine/itb_emoji.py#L87

```
DATADIR = os.path.join(os.path.dirname(__file__), '../data')
# USER_DATADIR will be “~/.local/share/ibus-typing-booster/data” by default
USER_DATADIR = itb_util.xdg_save_data_path('ibus-typing-booster/data')
CLDR_ANNOTATION_DIRNAMES = (
    USER_DATADIR, DATADIR,
    # On Fedora >= 25 there is a
    # “cldr-emoji-annotation” package which has the
    # .xml files here in the subdirs “annotations”
    # and “annotationsDerived”:
    '/usr/share/unicode/cldr/common/',
    '/local/mfabian/src/cldr/common/')
```

(I just notice I should remove `/local/mfabian/src/cldr/common/`,  this is never used anyway).

But as the test cases are only expected to work when the CLDR annotation files included with ibus-typing-booster are used, I try to skip the these tests when the CLDR annotation files are loaded from some other place. Like this:


https://github.com/mike-fabian/ibus-typing-booster/blob/main/tests/test_emoji_candidates.py#L54

```
@unittest.skipIf(
    '..' not in itb_emoji.find_cldr_annotation_path('en'),
    'Using external emoji annotations: %s '
    % itb_emoji.find_cldr_annotation_path('en')
    + 'Testing with older emoji annotations instead '
    'of those included in the ibus-typing-booster source is likely '
    'to create meaningless test failures.')
class EmojiCandidatesTestCase(unittest.TestCase):
```

So when I am in the top level directory of the ibus-typing-booster source and run `make check` (or in the tests directory and run `./test_emoji_candidates.py -v`), then `itb_emoji.find_cldr_annotation_path('en')` returns something like `/local/mfabian/src/ibus-typing-booster/tests/../engine/../data/annotations/en.xml` which contains `..` so the condition

```
'..' not in itb_emoji.find_cldr_annotation_path('en')
```

is false and the test is run.

During a rpm build for Fedora, the test `test_emoji_candidates.py` is run as well because the CLDR annotation files are still there in the source directory during the build and in CLDR_ANNOTATION_DIRNAMES above, DATADIR has higher priority than `/usr/share/unicode/cldr/common/`.

The final rpm packages on Fedora are:

```
emoji-picker-2.14.10-1.fc34.noarch.rpm
ibus-typing-booster-2.14.10-1.fc34.noarch.rpm
ibus-typing-booster-2.14.10-1.fc34.src.rpm
ibus-typing-booster-tests-2.14.10-1.fc34.noarch.rpm
```

The -tests package contains the test cases:

```
$ rpm -qlp ibus-typing-booster-tests-2.14.10-1.fc34.noarch.rpm 
/usr/libexec/installed-tests
/usr/libexec/installed-tests/ibus-typing-booster
/usr/libexec/installed-tests/ibus-typing-booster/gtkcases.py
/usr/libexec/installed-tests/ibus-typing-booster/mock_engine.py
/usr/libexec/installed-tests/ibus-typing-booster/run_tests
/usr/libexec/installed-tests/ibus-typing-booster/test_0_gtk.py
/usr/libexec/installed-tests/ibus-typing-booster/test_compose_sequences.py
/usr/libexec/installed-tests/ibus-typing-booster/test_emoji_candidates.py
/usr/libexec/installed-tests/ibus-typing-booster/test_emoji_similar.py
/usr/libexec/installed-tests/ibus-typing-booster/test_emoji_unicode_version.py
/usr/libexec/installed-tests/ibus-typing-booster/test_hunspell_suggest.py
/usr/libexec/installed-tests/ibus-typing-booster/test_itb.py
/usr/libexec/installed-tests/ibus-typing-booster/test_itb_util.py
/usr/libexec/installed-tests/ibus-typing-booster/test_keyvals_to_keycodes.py
/usr/libexec/installed-tests/ibus-typing-booster/test_m17n_db_info.py
/usr/libexec/installed-tests/ibus-typing-booster/test_m17n_translit.py
/usr/share/installed-tests
/usr/share/installed-tests/ibus-typing-booster
/usr/share/installed-tests/ibus-typing-booster/test_0_gtk.test
/usr/share/installed-tests/ibus-typing-booster/test_compose_sequences.test
/usr/share/installed-tests/ibus-typing-booster/test_emoji_candidates.test
/usr/share/installed-tests/ibus-typing-booster/test_emoji_similar.test
/usr/share/installed-tests/ibus-typing-booster/test_emoji_unicode_version.test
/usr/share/installed-tests/ibus-typing-booster/test_hunspell_suggest.test
/usr/share/installed-tests/ibus-typing-booster/test_itb.test
/usr/share/installed-tests/ibus-typing-booster/test_itb_util.test
/usr/share/installed-tests/ibus-typing-booster/test_keyvals_to_keycodes.test
/usr/share/installed-tests/ibus-typing-booster/test_m17n_db_info.test
/usr/share/installed-tests/ibus-typing-booster/test_m17n_translit.test
$ cat /usr/share/installed-tests/ibus-typing-booster/test_emoji_candidates.test
[Test]
Type=session
Exec=/usr/libexec/installed-tests/ibus-typing-booster/run_tests test_emoji_candidates.py
Output=TAP
$ 
```

These are used in the Fedora CI system to run the test cases again after installation of the rpms.
Something like this is done during the CI test:

- VM is installed
- ibus-typing-booster packages and dependencies are installed
- init 3
- log in as root
- run `ibus-desktop-testing-runner --runner=gnome --tests=ibus-typing-booster --output=ibus-typing-booster.log --result=test.log`

In this case, the CLDR emoji annotations are loaded from `/usr/share/unicode/cldr/common/` because they are not in the ibus-typing-booster.rpm, then `itb_emoji.find_cldr_annotation_path('en')` returns `/usr/share/unicode/cldr/common/annotations/en.xml` which contains no `..` and the emoji tests are skipped.

OK, so now I wonder why this skipping doesn’t work in your case.
From where are the emoji annotations actually loaded in when it fails for you?

I’ll soon release ibus-typing-booster-2.14.12 with a tiny bugfix for the German neo2 layout (https://github.com/mike-fabian/ibus-typing-booster/issues/243).

I’ll include extra logging output like this:

```
$ ./test_emoji_candidates.py -v
test_candidates_animals (__main__.EmojiCandidatesTestCase) ... itb_emoji.find_cldr_annotation_
path('en')->/local/mfabian/src/ibus-typing-booster/tests/../engine/../data/annotations/en.xml
ok
```

So then we can see in your next test-suite.log file from where these annotations are loaded when your tests run and why skipping if it doesn’t contain `..` doesn’t work.

**Comment by gunnarhj** [comment]
Thanks for your explanations!

To clarify: I'm not talking about CI tests after installation, neither am I talking about "autopkgtest" which is a similar thing on Debian to run tests on the installed packages and where passing is a condition for a built package to migrate to the pocket for published packages. I have not (yet) enabled any of those features for `ibus-typing-booster`, so I'm only talking about running the test suite during the build from source.

> This saves some space because ibus uses this CLDR emoji annotation data as well and uses thouse from the cldr-emoji-annotation-39-1.fc34.noarch package which therefore must be installed.

Hmm.. Does ibus on Fedora have a runtime dependency on `cldr-emoji-annotation-39-1.fc34.noarch`? We have no equivalent in Debian AFAIK. Should we?

Debian/Ubuntu has a `unicode-cldr-core` package. So I understand that an option would be to not include the `annotations/` and `annotationsDerived/` files when building the `ibus-typing-booster` `.deb` package, and instead depend on `unicode-cldr-core`. However, [the latter is huge](https://packages.ubuntu.com/impish/all/unicode-cldr-core/filelist), and currently only one package (`fcitx5-module-emoji`) depends on it at runtime.

> I’ll soon release ibus-typing-booster-2.14.12 with a tiny bugfix for the German neo2 layout (#243).
>
> I’ll include extra logging output like this:
>
>     $ ./test_emoji_candidates.py -v
>     test_candidates_animals (__main__.EmojiCandidatesTestCase) ... itb_emoji.find_cldr_annotation_
>     path('en')->/local/mfabian/src/ibus-typing-booster/tests/../engine/../data/annotations/en.xml
>     ok
>
> So then we can see in your next test-suite.log file from where these annotations are loaded when your tests run and why skipping if it doesn’t contain `..` doesn’t work.

That will be interesting to see.

**Comment by mike-fabian** [comment]
> > This saves some space because ibus uses this CLDR emoji annotation data as well and uses thouse from the cldr-emoji-annotation-39-1.fc34.noarch package which therefore must be installed.
> 
> Hmm.. Does ibus on Fedora have a runtime dependency on `cldr-emoji-annotation-39-1.fc34.noarch`?

Yes, the ibus-typing-booster.rpm on Fedora has a runtime dependency on the cldr-emoji-annotation. The reason is to share these data between ibus and ibus-typing-booster and save some space by not making both packages include the same data.

> We have no equivalent in Debian AFAIK. Should we?

I don’t know. Including the annotation data from the ibus-typing-booster source  in the ibus-typing-booster package will ususally give more up-to-date annotation data. The `cldr-emoji-annotation` package on Fedora is updated rarely (I don’t maintain it). The cldr annotation data in the ibus-typing-booster source is updated to git master on almost every release of ibus-typing-booster.

The unzipped cldr annotation data is around 80 megabytes:

`cldr-emoji-annotation` package:

```
$ du -s -h -c  /usr/share/unicode/cldr/common/annotations*
32M	/usr/share/unicode/cldr/common/annotations
47M	/usr/share/unicode/cldr/common/annotationsDerived
78M	total
```

ibus-typing-booster source:

```
mfabian@taka:/local/mfabian/src/ibus-typing-booster (release-candidate-2.14.12 $)
$ du -s -h -c data/annotations*
36M     data/annotations
49M     data/annotationsDerived
84M     total
mfabian@taka:/local/mfabian/src/ibus-typing-booster (release-candidate-2.14.12 $)
$ 
```

I don’t know why the `cldr-emoji-annotation` package doesn’t gzip the files, maybe ibus needs them unzipped? I am not sure whether ibus needs these files at runtime at all or only at build time, I should check.

ibus-typing-booster looks for `.xml` first and if that is not found it looks for `.xml.gz`:

https://github.com/mike-fabian/ibus-typing-booster/blob/main/engine/itb_emoji.py#L256

Therefore, when including these files in the ibus-typing-booster package, it is good to gzip them, this saves a lot of space. If you don’t do that already I would recommend to gzip the xml files.

```
$ du -s -h -c data/annotations*
5.4M    data/annotations
3.4M    data/annotationsDerived
8.8M    total
```

> Debian/Ubuntu has a `unicode-cldr-core` package. So I understand that an option would be to not include the `annotations/` and `annotationsDerived/` files when building the `ibus-typing-booster` `.deb` package, and instead depend on `unicode-cldr-core`. However, [the latter is huge](https://packages.ubuntu.com/impish/all/unicode-cldr-core/filelist), and currently only one package (`fcitx5-module-emoji`) depends on it at runtime.

Yes, it would be possible to require that `unicode-cldr-core` package and not include the cldr emoji annotations in the ibus-typing-booster.deb package.

The location of the files is apparently the same as in the Fedora `cldr-emoji-annotation` package, so the current code in ibus-typing-booster to search for these files should also work on Debian if the files are not found first in the /usr/share/ibus-typing-booster/data/ directory.

In the package for openSUSE, I include the gzipped .xml files in the ibus-typing-booster.rpm.

I don’t know what to recommend, is it better to have the most up-to-date files in the ibus-typing-booster package and make the package 9 megabytes bigger (gzipped),
or is it better to make the ibus-typing-booster package smaller and require some huge external package?

**Comment by mike-fabian** [comment]
> I don’t know why the `cldr-emoji-annotation` package doesn’t gzip the files, maybe ibus needs them unzipped? I am not sure whether ibus needs these files at runtime at all or only at build time, I should check.

I checked, ibus reads those files only at build time and converts them into some internal format, it doesn’t use them at runtime.

ibus-typing-booster reads these files at runtime during startup.

**Comment by gunnarhj** [comment]
>>> This saves some space because ibus uses this CLDR emoji annotation data as well and uses thouse from the cldr-emoji-annotation-39-1.fc34.noarch package which therefore must be installed.
>>
>>  Hmm.. Does ibus on Fedora have a runtime dependency on `cldr-emoji-annotation-39-1.fc34.noarch`?
>
> Yes, the ibus-typing-booster.rpm on Fedora has a runtime dependency on the cldr-emoji-annotation. 

Yes, but what about ibus itself? That's what I meant to ask about.

> I checked, ibus reads those files only at build time and converts them into some internal format, it doesn’t use them at runtime.

That's what I thought, and the reason why I found it hard to follow your talk about avoiding duplication.

> ibus-typing-booster looks for .xml first and if that is not found it looks for .xml.gz:
>
> https://github.com/mike-fabian/ibus-typing-booster/blob/main/engine/itb_emoji.py#L256
>
> Therefore, when including these files in the ibus-typing-booster package, it is good to gzip them, this saves a lot of space. If you don’t do that already I would recommend to gzip the xml files.

Thanks for the tip! Will try that.

> I don’t know what to recommend, is it better to have the most up-to-date files in the ibus-typing-booster package and make the package 9 megabytes bigger (gzipped),
> or is it better to make the ibus-typing-booster package smaller and require some huge external package?

For Debian/Ubuntu I think the former is preferable, since the external package is large and not installed for other reasons by most users.

**Comment by mike-fabian** [comment]
> > I’ll soon release ibus-typing-booster-2.14.12 with a tiny bugfix for the German neo2 layout (#243).
> > I’ll include extra logging output like this:
> > ```
> > $ ./test_emoji_candidates.py -v
> > test_candidates_animals (__main__.EmojiCandidatesTestCase) ... itb_emoji.find_cldr_annotation_
> > path('en')->/local/mfabian/src/ibus-typing-booster/tests/../engine/../data/annotations/en.xml
> > ok
> > ```
> >   
> > So then we can see in your next test-suite.log file from where these annotations are loaded when your tests run and why skipping if it doesn’t contain `..` doesn’t work.
> 
> That will be interesting to see.

I have release ibus-typing-booster-2.14.12 with that extra logging which should show from where the CLDR annotation files are loaded:

https://github.com/mike-fabian/ibus-typing-booster/releases/tag/2.14.12

**Comment by mike-fabian** [comment]
By the way, the *.txt files can be zipped as well to make the package smaller:

/usr/share/ibus-typing-booster/data/UnicodeData.txt
/usr/share/ibus-typing-booster/data/emoji-data.txt
/usr/share/ibus-typing-booster/data/emoji-sequences.txt
/usr/share/ibus-typing-booster/data/emoji-test.txt
/usr/share/ibus-typing-booster/data/emoji-variation-sequences.txt
/usr/share/ibus-typing-booster/data/emoji-zwj-sequences.txt

**Comment by gunnarhj** [comment]
> I have released ibus-typing-booster-2.14.12 with that extra logging which should show from where the CLDR annotation files are loaded:

Tried it, but no additional info was printed to [test-suite.log](https://github.com/mike-fabian/ibus-typing-booster/files/7181808/test-suite.log).

**Comment by mike-fabian** [comment]
Ah, you are right, my new log output is only printed if I run the individual test directly from the command line!

Like this:

```
$ ./test_emoji_candidates.py -v
test_candidates_animals (__main__.EmojiCandidatesTestCase) ... itb_emoji.find_cldr_annotation_
path('en')->/local/mfabian/src/ibus-typing-booster/tests/../engine/../data/annotations/en.xml
ok
```

I thought it would go into the normal log as well, but I was mistaken.

You wrote:

> I noticed a failure with respect to tests/test_emoji_candidates.py when building locally not in a clean chroot, i.e. in an environment with a lot of installed packages which are irrelevant for the package build.

In such a chroot where it fails, can you go to the directory `ibus-typing-booster/tests` in the sources of ibus-typing-booste
and run `./test_emoji_candidates.py -v` there directly?

Then it should print to standard output the path where the emoji annotations are loaded from

**Comment by gunnarhj** [comment]
> In such a chroot where it fails, can you go to the directory `ibus-typing-booster/tests` in the sources of ibus-typing-booste
and run `./test_emoji_candidates.py -v` there directly?

Did that, and this is the result:

[output-when-run-test-from-command-line.txt](https://github.com/mike-fabian/ibus-typing-booster/files/7188774/output-when-run-test-from-command-line.txt)

Also the failed tests seem to be based on `data/annotations/en.xml` in the source.

**Comment by mike-fabian** [comment]
Weird. I just downloaded ubuntu-20.04.3-desktop-amd64.iso and installed it in a VM.

Then installed the build and runtime dependencies and tools, and then did

```
git clone https://github.com/mike-fabian/ibus-typing-booster.git
cd ibus-typing-booster
./autogen.sh --prefix=/usr --libexecdir=/usr/lib/ibus
make
make dist
cd tests
./test_emoji_candidates.py -v 
```

and the test worked without errors.

**Comment by mike-fabian** [comment]
Is there anything in this directory?:

```
mfabian@ubuntu-20:~/ibus-typing-booster/tests$ ls ~/.local/share/ibus-typing-booster/data/
mfabian@ubuntu-20:~/ibus-typing-booster/tests$ 
```

**Comment by mike-fabian** [comment]
And does python3-enchant work correctly on your system? I.e. something like this works?:

```
mfabian@ubuntu-20:~/ibus-typing-booster/tests$ python3
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import enchant
>>> d = enchant.Dict('en')
>>> d.check('fery')
False
>>> d.suggest('fery')
['fer', 'fey', 'fry', 'fiery', 'ferny', 'ferry', 'fern', 'fury', 'very', 'fer y']
>>> 
```

**Comment by gunnarhj** [comment]
Your latest comment lead me in the right direction.

```
gunnar@gunnar-ubuntu-dev:~$ python3
Python 3.9.7 (default, Sep 10 2021, 14:59:43) 
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import enchant
>>> d = enchant.Dict('en')
>>> d.check('fery')
False
>>> d.suggest('fery')
['fer', 'fey', 'fry', 'fiery', 'faery', 'ferny', 'ferry', 'fern', 'fury', 'very', 'fer y', 'fairy', 'frumpy']
>>> quit()
```

I got a longer suggestion list than you, though. That's because I have multiple English hunspell dictionaries installed, and it seems to take into account all of them. So:

```
Python 3.9.7 (default, Sep 10 2021, 14:59:43) 
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import enchant
>>> d = enchant.Dict('en_US')
>>> d.suggest('fery')
['Frey', 'ferry', 'fiery', 'Fry', 'ferny', 'fry', 'fer', 'fey', 'fury', 'furry', 'fray', 'Fern', 'fern', 'very', 'Fr', 'fr', 'free', 'fairy', 'foray', 'Fri', 'fro', 'fare', 'fire', 'fore', 'faro', 'fora']
>>> d.suggest('mmgb')
['MGM', 'Mugabe', 'GB', 'MB', 'Mb', 'Mg', 'mg', 'Meg', 'MiG', 'Mujib', 'mag', 'meg', 'mob', 'mug', 'KGB', 'Mgr', 'mgr', 'MEGO', 'Magi', 'mage', 'magi', 'mega', 'mags', 'megs', 'mugs', 'mamba', 'mambo', 'MBA', 'gab', 'gob', 'magma', 'Moog', "Mg's", 'CB', 'Cb', 'KB', 'Kb', 'MC', 'Mk', 'QB', "Meg's", "MiG's", "mag's", "mug's"]
>>> quit()
```

Oops! When excluding hunspell dicts such as en_GB and en_CA it suddenly got very creative. Note: "free" as in "squared free" and "mag" as in "left-pointing magnifying glass".

I had the `aspell-en` package installed, and for some reason it has started to take the aspell en_US dict into account when generating the suggestion list. It didn't do so in Ubuntu 21.04 (and probably not in 20.04 either), but it does in Ubuntu 21.10 and in Debian testing. One common denominator for the two latter is that the python3 version is 3.9.7 (in 21.04 it's 3.9.5).

Uninstalling `aspell-en` results in:

```
gunnar@gunnar-ubuntu-dev:~$ python3
Python 3.9.7 (default, Sep 10 2021, 14:59:43) 
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import enchant
>>> d = enchant.Dict('en_US')
>>> d.suggest('fery')
['fer', 'fey', 'fry', 'fiery', 'ferny', 'ferry', 'fern', 'fury', 'very', 'fer y']
>>> d.suggest('mmgb')
['mg']
>>> quit()
```

and `ibus-typing-booster` can be built successfully.

So now we know the reason for my reported test failures when building locally. The question is whether this new enchant behavior will become a runtime issue. Maybe not. A longer suggestion list isn't worse per se compared to a shorter list, I suppose.

**Comment by mike-fabian** [comment]
No, a longer list of words from the spellchecker is not necessarily bad. 

I just wonder how I can change the test case(s) to not  produce such unhelpful failures.

As I skip the emoji tests already when the annotation files are loaded from outside the source tree, maybe I should also check the results of enchant with a few simple examples and if they are different from what I expect, don’t report a failure but skip the emoji matching tests. Probably these emoji matching tests  don’t have to run on every system out there, if they run on the system I currently use for development, that is probably good enough for me to notice if I break the emoji matching with some change.
And if they are suddenly all skipped because the behaviour of enchant changes (which it probably will at some point in the future on my system as well), then I can adapt the test cases.

**Comment by gunnarhj** [comment]
> Probably these emoji matching tests don’t have to run on every system out there, if they run on the system I currently use for development, that is probably good enough for me to notice if I break the emoji matching with some change.

Agreed. Which speaks for not putting too much effort in changing the test cases back and forth. As regards Debian/Ubuntu, and for now, I can simply make sure that `aspell-en` is not installed before doing test builds locally.

Thanks for helping identity the cause of the issue! And thanks especially for the gzip'ing tip. I think the latter in itself motivates an additional upload.
